engine/src/execution_context.rs

```diff
@@ -69,6 +70,60 @@ impl<'e> ExecutionContext<'e> {
             })
         }
     }
+
+    /// Sets a runtime value for a given field name and a path
+    pub fn set_field_value_with_path<'v: 'e, V: Into<LhsValue<'v>>>(
```

Comment:
Elie and I discussed that we will rename this from path to indexes to be coherent with FieldIndex.